### PR TITLE
Issue 52 - Added Telegram plugin

### LINKED ISSUE #52: add Telegram sharing
Along with the "Share with WhatsApp" Button for mobile, it would be nice to have a "Share with Telegram" Button (mind the desktop app)

(see also here https://github.com/DrKLO/Telegram/issues/652)

### PR DESCRIPTION
**Pull request by rodfersou**
closes #52

### CODE REVIEW

File: sc/social/like/plugins/telegram/browser.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        self.portal_state = getMultiAdapter(
+            (self.context, self.request), name=u'plone_portal_state')
+        self.portal = self.portal_state.portal()
+        self.site_url = self.portal_state.portal_url()
+        self.portal_title = self.portal_state.portal_title()
+        self.url = self.context.absolute_url()
+        self.language = get_language(self.context)
+        self.telegramurl = (
+            u'https://telegram.me/share/url?url=' +
+            url_quote(
```
Comment (hvelarde) [inline sc/social/like/plugins/telegram/browser.py]: I think is better to leave this in one line

File: sc/social/like/plugins/telegram/configure.zcml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<configure
+    xmlns="http://namespaces.zope.org/zope"
+    xmlns:browser="http://namespaces.zope.org/browser"
+    i18n_domain="sc.social.like">
+
+  <!-- Register Plugin -->
+  <utility
```
Comment (hvelarde) [inline sc/social/like/plugins/telegram/configure.zcml]: I think is better to leave this in one line.

File: sc/social/like/tests/test_plugin_telegram.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+    def test_plugin_urlnoscript_encoding(self):
+        plugin = self.plugin
+        document = self.document
+        document.setTitle(u'Notícia')
+
+        plugin_view = plugin.view()
+        view = document.restrictedTraverse(plugin_view)
+        html = view.plugin()
+        self.assertIn(
```
Comment (hvelarde) [inline sc/social/like/tests/test_plugin_telegram.py]: I think is better to leave this in one line.

File: sc/social/like/tests/test_plugin_telegram.robot [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+*** Variables ***
+
+${title_selector} =  input#title
+${telegram_locator}  css=#viewlet-social-like .telegram
+
+*** Test cases ***
+
+Test Telegram
+    [Tags]  Expected Failure
```
Comment (hvelarde) [inline sc/social/like/tests/test_plugin_telegram.robot]: why is this an Expected Failure?

File: sc/social/like/tests/test_plugin_telegram.py [hunk 5]
```diff
@@ -54,38 +53,29 @@ class PluginViewsTest(unittest.TestCase):
...
 
     def test_plugin_view(self):
-        plugin = self.plugin
-        portal = self.portal
-        plugin_view = plugin.view()
-        view = portal.restrictedTraverse(plugin_view)
-        self.assertTrue(isinstance(view, browser.PluginView))
+        plugin = self.plugin.view()
+        view = self.portal.restrictedTraverse(plugin)
+        self.assertTrue(isinstance(view, PluginView))
```
Comment (rodfersou) [inline sc/social/like/tests/test_plugin_telegram.py]: this could be `self.assertIsInstance(view, PluginView)` instead

**Pull request merged**